  <main role="main">    <h1>Class WriteAbortedException</h1>   <ul class="inheritance"> <li><a href="../lang/object">java.lang.Object</a></li> <li> <ul class="inheritance"> <li><a href="../lang/throwable">java.lang.Throwable</a></li> <li> <ul class="inheritance"> <li><a href="../lang/exception">java.lang.Exception</a></li> <li> <ul class="inheritance"> <li><a href="ioexception">java.io.IOException</a></li> <li> <ul class="inheritance"> <li><a href="objectstreamexception">java.io.ObjectStreamException</a></li> <li> <ul class="inheritance"> <li>java.io.WriteAbortedException</li> </ul> </li> </ul> </li> </ul> </li> </ul> </li> </ul> </li> </ul>   <dl> <dt>All Implemented Interfaces:</dt> <dd><code><a href="serializable">Serializable</a></code></dd> </dl>  <pre data-language="java">public class WriteAbortedException
extends ObjectStreamException</pre> <div>
<p>Signals that one of the ObjectStreamExceptions was thrown during a write operation. Thrown during a read operation when one of the ObjectStreamExceptions was thrown during a write operation. The exception that terminated the write can be found in the detail field. The stream is reset to it's initial state and all references to objects already deserialized are discarded. </p>
<p>As of release 1.4, this exception has been retrofitted to conform to the general purpose exception-chaining mechanism. The "exception causing the abort" that is provided at construction time and accessed via the public <a href="#detail"><code>detail</code></a> field is now known as the <i>cause</i>, and may be accessed via the <a href="../lang/throwable#getCause()"><code>Throwable.getCause()</code></a> method, as well as the aforementioned "legacy field."</p>
</div> <dl> <dt>Since:</dt> <dd>1.1</dd> <dt>See Also:</dt> <dd><a href="https://docs.oracle.com/en/java/javase/11/docs/api/serialized-form.html#java.io.WriteAbortedException">Serialized Form</a></dd> </dl>      <section role="region">  <h2 id="field.summary">Fields</h2> <table>  <tr> <th scope="col">Modifier and Type</th> <th scope="col">Field</th> <th scope="col">Description</th> </tr> <tr> <td><code><a href="../lang/exception">Exception</a></code></td> <th scope="row"><code><a href="#detail">detail</a></code></th> <td> <p>Exception that was caught while writing the ObjectStream.</p> </td> </tr> </table>  </section>  <section role="region">  <h2 id="constructor.summary">Constructors</h2> <table>  <tr> <th scope="col">Constructor</th> <th scope="col">Description</th> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.String,java.lang.Exception)">WriteAbortedException</a>​(<a href="../lang/string">String</a> s,
                     <a href="../lang/exception">Exception</a> ex)</code></th> <td> <p>Constructs a WriteAbortedException with a string describing the exception and the exception causing the abort.</p> </td> </tr> </table>  </section>  <section role="region">  <h2 id="method.summary">Methods</h2> <table>  <tr> <th scope="col">Modifier and Type</th> <th scope="col">Method</th> <th scope="col">Description</th> </tr> <tr id="i0"> <td><code><a href="../lang/throwable">Throwable</a></code></td> <th scope="row"><code><a href="#getCause()">getCause</a>()</code></th> <td> <p>Returns the exception that terminated the operation (the <i>cause</i>).</p> </td> </tr> <tr id="i1"> <td><code><a href="../lang/string">String</a></code></td> <th scope="row"><code><a href="#getMessage()">getMessage</a>()</code></th> <td> <p>Produce the message and include the message from the nested exception, if there is one.</p> </td> </tr> </table>  <h2 id="methods.inherited.from.class.java.lang.Throwable">Methods declared in class java.lang.<a href="../lang/throwable">Throwable</a>
</h2> <code><a href="../lang/throwable#addSuppressed(java.lang.Throwable)">addSuppressed</a>, <a href="../lang/throwable#fillInStackTrace()">fillInStackTrace</a>, <a href="../lang/throwable#getLocalizedMessage()">getLocalizedMessage</a>, <a href="../lang/throwable#getStackTrace()">getStackTrace</a>, <a href="../lang/throwable#getSuppressed()">getSuppressed</a>, <a href="../lang/throwable#initCause(java.lang.Throwable)">initCause</a>, <a href="../lang/throwable#printStackTrace()">printStackTrace</a>, <a href="../lang/throwable#printStackTrace(java.io.PrintStream)">printStackTrace</a>, <a href="../lang/throwable#printStackTrace(java.io.PrintWriter)">printStackTrace</a>, <a href="../lang/throwable#setStackTrace(java.lang.StackTraceElement%5B%5D)">setStackTrace</a>, <a href="../lang/throwable#toString()">toString</a></code>  <h2 id="methods.inherited.from.class.java.lang.Object">Methods declared in class java.lang.<a href="../lang/object">Object</a>
</h2> <code><a href="../lang/object#clone()">clone</a>, <a href="../lang/object#equals(java.lang.Object)">equals</a>, <a href="../lang/object#finalize()">finalize</a>, <a href="../lang/object#getClass()">getClass</a>, <a href="../lang/object#hashCode()">hashCode</a>, <a href="../lang/object#notify()">notify</a>, <a href="../lang/object#notifyAll()">notifyAll</a>, <a href="../lang/object#wait()">wait</a>, <a href="../lang/object#wait(long)">wait</a>, <a href="../lang/object#wait(long,int)">wait</a></code>  </section>   <div>   <section role="region">  <h2 id="field.detail">Fields</h2>   <h3 id="detail">detail</h3> <pre data-language="java">public Exception detail</pre> <div>
<p>Exception that was caught while writing the ObjectStream. </p>
<p>This field predates the general-purpose exception chaining facility. The <a href="../lang/throwable#getCause()"><code>Throwable.getCause()</code></a> method is now the preferred means of obtaining this information.</p>
</div>   </section>  <section role="region">  <h2 id="constructor.detail">Constructors</h2>   <h3 id="&lt;init&gt;(java.lang.String,java.lang.Exception)">WriteAbortedException</h3> <pre data-language="java">public WriteAbortedException(String s,
                             Exception ex)</pre> <p>Constructs a WriteAbortedException with a string describing the exception and the exception causing the abort.</p> <dl> <dt>Parameters:</dt> <dd>
<code>s</code> - String describing the exception.</dd> <dd>
<code>ex</code> - Exception causing the abort.</dd> </dl>   </section>  <section role="region">  <h2 id="method.detail">Methods</h2>   <h3 id="getMessage()">getMessage</h3> <pre data-language="java">public String getMessage()</pre> <p>Produce the message and include the message from the nested exception, if there is one.</p> <dl> <dt>Overrides:</dt> <dd>
<code><a href="../lang/throwable#getMessage()">getMessage</a></code> in class <code><a href="../lang/throwable">Throwable</a></code>
</dd> <dt>Returns:</dt> <dd>the detail message string of this <code>Throwable</code> instance (which may be <code>null</code>).</dd> </dl>    <h3 id="getCause()">getCause</h3> <pre data-language="java">public Throwable getCause()</pre> <p>Returns the exception that terminated the operation (the <i>cause</i>).</p> <dl> <dt>Overrides:</dt> <dd>
<code><a href="../lang/throwable#getCause()">getCause</a></code> in class <code><a href="../lang/throwable">Throwable</a></code>
</dd> <dt>Returns:</dt> <dd>the exception that terminated the operation (the <i>cause</i>), which may be null.</dd> <dt>Since:</dt> <dd>1.4</dd> </dl>   </section>  </div>  </main>  <div class="_attribution">
  <p class="_attribution-p">
    &copy; 1993, 2020, Oracle and/or its affiliates. All rights reserved.<br>Documentation extracted from Debian's OpenJDK Development Kit package.<br>Licensed under the GNU General Public License, version 2, with the Classpath Exception.<br>Various third party code in OpenJDK is licensed under different licenses (see Debian package).<br>Java and OpenJDK are trademarks or registered trademarks of Oracle and/or its affiliates.<br>
    <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/WriteAbortedException.html" class="_attribution-link">https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/WriteAbortedException.html</a>
  </p>
</div>
